File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let minSize = Math.min(sizes[0][0], sizes[0][1]);
    let maxSize = Math.max(sizes[0][0], sizes[0][1]);

    for(let i = 1; i < sizes.length; i++){
        let currentMin = Math.min(sizes[i][0], sizes[i][1]);
        let currentMax = Math.max(sizes[i][0], sizes[i][1]);

        if (currentMin > minSize)  minSize = currentMin;
        if (currentMax > maxSize)  maxSize = currentMax;
    }

    return minSize * maxSize;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    max_width = 0
    min_height = 0
    
    for i in range(len(sizes)) :
        if max_width >= max(sizes[i]) and min_height >= min(sizes[i]) :
            continue;
        else :
            if max_width <= max(sizes[i]) : 
                max_width = max(sizes[i])  
            if min_height <= min(sizes[i]) :
                min_height = min(sizes[i])    
    answer = max_width * min_height
    return answer